src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -797,6 +803,7 @@ abstract class ManagedStrategy(
             null
         }
         newSwitchPoint(iThread, codeLocation, tracePoint)
+        loopDetector.beforeWriteField(obj, value)
```

Comment:
Should it be called before newSwitchPoint or after? Why?

Reply:
In the current implementation it doesn't matter. But it looks more logical after the `newSwitchPoint` call as after we back to a thread we want to trace field read after the switch, not before (as it will actually happen).